ethstore/src/account/safe_account.rs

```diff
@@ -76,17 +76,28 @@ impl SafeAccount {
 
 	/// Create a new `SafeAccount` from the given `json`; if it was read from a
 	/// file, the `filename` should be `Some` name. If it is as yet anonymous, then it
-	/// can be left `None`.
-	pub fn from_file(json: json::KeyFile, filename: Option<String>) -> Self {
-		SafeAccount {
+	/// can be left `None`. `password` is used for reading keystore files w/o address.
+	pub fn from_file(json: json::KeyFile, filename: Option<String>, password: &Option<Password>) -> Result<Self, Error> {
+		let crypto = Crypto::from(json.crypto);
+		let address = match json.address {
+			Some(address) => address.into(),
+			None => match password {
```

Comment:
what if the json file contains `address` and we provided a password? imo, this should result in an `Error`

Reply:
Again, I would silently ignore this. As for now all errors reading kestores are silently ignored, which is confusing me too, but to be consistent I propose to ignore it.